src/main/java/net/sf/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java

```diff
@@ -493,10 +504,10 @@ public static String applyModifiers(String label, List<String> parts, int offset
             for (int j = offset; j < parts.size(); j++) {
                 String modifier = parts.get(j);
 
-                if ("lower".equals(modifier)) {
-                    resultingLabel = resultingLabel.toLowerCase(Locale.ENGLISH);
-                } else if ("upper".equals(modifier)) {
-                    resultingLabel = resultingLabel.toUpperCase(Locale.ENGLISH);
+                if ("lower".equals(modifier) || lowerCaseFormatter.getKey().equals(modifier)) {
```

Comment:
Please refactor the code a bit: first convert the modifier to the corresponding formatter `Formatter getFormatterForModifier(string modifier)` and then apply it to the label.